src/literal.cc

```diff
@@ -718,6 +725,12 @@ Result parse_double(LiteralType literal_type,
                     uint64_t* out_bits) {
   switch (literal_type) {
     case LiteralType::Int:
+#if COMPILER_IS_MSVC
+      if (string_starts_with(s, end, "0x"))
+      {
+        goto parseDoubleHex;
```

Comment:
using `goto` here seems little extreme.  Could you instead do something like this before the switch?

```
#if COMPILER_IS_MSVC
if (literal_type == LiteralType::Int && string_starts_with(s, end, "0x"))
  literal_type = LiteralType::Hexfloat
#endif
```
Also, this is probably worthy of comment explaining why.